alf/metrics/metrics.py

```diff
@@ -415,8 +400,9 @@ def __init__(self,
             name=name,
             dtype=dtype,
             prefix=prefix,
-            batch_size=batch_size,
-            buffer_size=buffer_size)
+            buffer_size=buffer_size,
+            example_time_step=TimeStep(
+                step_type=torch.full((batch_size, ), StepType.FIRST)))
```

Comment:
Maybe also pass in the example_time_step instead of creating a dummy one here? Also batch_size will no longer be needed as others. This will unify the constructors of all Metrics.

Reply:
Sounds good. Originally I was thinking about trading off consistency for simplicity, but either way is fine.